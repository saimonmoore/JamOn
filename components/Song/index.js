import React, { Component } from 'react';
import {
  Button,
  Text,
  View,
} from 'react-native';

import { observer } from 'mobx-react/native';
import autobind from 'autobind-decorator';
import Sessions from '../Sessions';

@observer
class Song extends Component {
  @autobind editSong() {
    const song = this.props.song;
    const { navigate } = this.props.navigation;

    if (song) {
      navigate('SongForm', { song });
    }
  }

  render() {
    const { navigation } = this.props;

    return (
      <View style={{flex: 1, paddingTop: 22}}>
        <View>
          <Text style={{ fontWeight: 'bold' }}>{this.props.song.name}</Text>
        </View>
        <View>
          <Text>{this.props.song.author}</Text>
        </View>
        <View>
          <Text>{this.props.song.genre}</Text>
        </View>
        <Button
          onPress={this.editSong}
          title="Edit"
          color="#841584"
          accessibilityLabel="Press to edit the song"
        />
        <View>
          <Text style={{ textDecorationLine: 'underline', color: 'red' }}>JamOn Sessions</Text>
        </View>
        <Sessions song={this.props.song} navigation={navigation} />
      </View>
    );
  }
}

export default Song;
