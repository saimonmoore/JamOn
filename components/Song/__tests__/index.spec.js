import 'react-native';
import React from 'react';
import Song from '../index.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('with a song', () => {
  const song = {name: 'Man of Constant Sorrow', author: 'Dick Burnett', genre: 'Traditional'};

  const component = renderer.create(
    <Song song={song} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
