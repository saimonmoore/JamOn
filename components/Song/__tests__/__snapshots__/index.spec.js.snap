exports[`test with a song 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 22,
    }
  }>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Man of Constant Sorrow
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Dick Burnett
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Traditional
    </Text>
  </View>
</View>
`;
